### PR TITLE
Use instance.external_name in secret.instance_name

### PR DESCRIPTION
**Pull request by TalShorSap**
Use instance.external_name in secret.instance_name

### CODE REVIEW

File: controllers/servicebinding_controller.go [hunk 1]
```diff
@@ -760,6 +759,16 @@ func (r *ServiceBindingReconciler) addInstanceInfo(ctx context.Context, binding
 	return metadata, nil
 }
 
+func getInstanceName(instance *servicesv1.ServiceInstance) []byte {
```
Comment (evyaffe) [inline controllers/servicebinding_controller.go:762]: getInstanceNameForSecretCredentials
(or something like that)

File: api/common.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ const (
 	StaleBindingRotationOfLabel string         = "services.cloud.sap.com/rotationOf"
 	ForceRotateAnnotation       string         = "services.cloud.sap.com/forceRotate"
 	PreventDeletion             string         = "services.cloud.sap.com/preventDeletion"
+	UseInstanceMetaNameInSecret string         = "services.cloud.sap.com/useMeteName"
```
Comment (evyaffe) [inline api/common.go:21]: reminder to use a less technical term (e.g use-kubernetes-instance-name-for-credentials)

File: controllers/servicebinding_controller_test.go [hunk 3]
```diff
@@ -516,6 +516,18 @@ var _ = Describe("ServiceBinding controller", func() {
...
 
+		Context("useMetaName annotation is provided", func() {
+			It("should put in the secret.instance_name the instance meta.name", func() {
+				createdInstance.Annotations = map[string]string{
+					api.UseInstanceMetaNameInSecret: "true",
+				}
+				updateInstance(ctx, createdInstance)
+				createdBinding = createBinding(ctx, bindingName, bindingTestNamespace, instanceName, "", "")
+				bindingSecret := getSecret(ctx, createdBinding.Spec.SecretName, createdBinding.Namespace, true)
+				validateSecretData(bindingSecret, "instance_name", instanceName)
```
Comment (evyaffe) [inline controllers/servicebinding_controller_test.go:527]: move validateInstanceInfo and validateSecretMetadata out of the "sync" context - create them as functions next to validateSecretData. 
Then in this test do  validateInstanceInfo and validateSecretMetadata. You want to make sure your changes (and future changes to that area) don't affect other unintended properties.

Comment (TalShorSap) [reply controllers/servicebinding_controller_test.go]: done

File: controllers/servicebinding_controller.go (hunk 1 above)
Comment (TalShorSap) [reply controllers/servicebinding_controller.go]: done

File: controllers/servicebinding_controller.go [hunk 4]
```diff
@@ -760,6 +759,16 @@ func (r *ServiceBindingReconciler) addInstanceInfo(ctx context.Context, binding
...
+	if useMetaName, ok := instance.Annotations[api.UseInstanceMetaNameInSecret]; ok {
```
[the remark this answers is not in the document]
Comment (TalShorSap) [reply controllers/servicebinding_controller.go]: done

File: api/common.go (hunk 2 above)
Comment (TalShorSap) [reply api/common.go:21]: Decided with Avi and Daniel
UseInstanceMetaDataName

**Review by evyaffe — APPROVED** [review]

**Pull request merged**